src/rulesets/validation.ts

```diff
@@ -0,0 +1,15 @@
+import AJV = require('ajv');
+import { ErrorObject } from 'ajv';
+import ruleSchema = require('../meta/rule.schema.json');
+import rulesetSchema = require('../meta/ruleset.schema.json');
+import { IRuleset } from '../types/ruleset';
+
+const ajv = new AJV({ allErrors: true, jsonPointers: true });
+const validate = ajv.addSchema(ruleSchema).compile(rulesetSchema);
+
+export function validateRuleset(ruleset: IRuleset): ErrorObject[] {
+  if (validate(ruleset)) {
```

Comment:
This seems to be doing three things? If its valid, then return [], otherwise try and return errors but also maybe return []. How is something going to be invalid but not have any errors? Is this just some hackery to make TS happy?

Reply:
> Is this just some hackery to make TS happy?

yup, errors can be `undefined`

Reply:
Wait why?!

Reply:
Because that's the return type :)